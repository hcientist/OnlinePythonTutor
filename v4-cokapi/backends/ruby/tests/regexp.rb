x = /^<hello> (.*)$/
y = Regexp.new('\<d+> (\d+)$')
