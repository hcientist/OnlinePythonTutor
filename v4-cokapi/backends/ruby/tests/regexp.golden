{
  "code": "x = /^<hello> (.*)$/\ny = Regexp.new('\\<d+> (\\d+)$')\n",
  "trace": [
    {
      "func_name": "<toplevel>",
      "line": 1,
      "event": "step_line",
      "stdout": "",
      "stack_to_render": [
        {
          "is_highlighted": true,
          "func_name": "<main>:1",
          "frame_id": 1,
          "unique_hash": "<main>:1_f1",
          "is_parent": false,
          "is_zombie": false,
          "parent_frame_id_list": [

          ],
          "ordered_varnames": [
            "x",
            "y"
          ],
          "encoded_locals": {
            "x": null,
            "y": null
          }
        }
      ],
      "heap": {
      },
      "globals": {
      },
      "ordered_globals": [

      ]
    },
    {
      "func_name": "<toplevel>",
      "line": 2,
      "event": "step_line",
      "stdout": "",
      "stack_to_render": [
        {
          "is_highlighted": true,
          "func_name": "<main>:2",
          "frame_id": 1,
          "unique_hash": "<main>:2_f1",
          "is_parent": false,
          "is_zombie": false,
          "parent_frame_id_list": [

          ],
          "ordered_varnames": [
            "x",
            "y"
          ],
          "encoded_locals": {
            "x": [
              "REF",
              1
            ],
            "y": null
          }
        }
      ],
      "heap": {
        "1": [
          "Regexp",
          "/^<hello> (.*)$/"
        ]
      },
      "globals": {
      },
      "ordered_globals": [

      ]
    },
    {
      "func_name": "<toplevel>",
      "line": 2,
      "event": "step_line",
      "stdout": "",
      "stack_to_render": [
        {
          "is_highlighted": true,
          "func_name": "<main>:3",
          "frame_id": 1,
          "unique_hash": "<main>:3_f1",
          "is_parent": false,
          "is_zombie": false,
          "parent_frame_id_list": [

          ],
          "ordered_varnames": [
            "x",
            "y"
          ],
          "encoded_locals": {
            "x": [
              "REF",
              1
            ],
            "y": [
              "REF",
              2
            ]
          }
        }
      ],
      "heap": {
        "1": [
          "Regexp",
          "/^<hello> (.*)$/"
        ],
        "2": [
          "Regexp",
          "/\\<d+> (\\d+)$/"
        ]
      },
      "globals": {
      },
      "ordered_globals": [

      ]
    }
  ]
}
